### PR TITLE
[CEDS-2786] Add db migration feature

### PR DESCRIPTION
**Pull request by glidester**
Adding a db migration feature that runs on application
startup as a precursor to making certain fields of the
Notification entity optional.

No migrations are included in this commit (will be
added later)

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -43,7 +43,9 @@ lazy val microservice = (project in file("."))
 val compileDependencies = Seq(
   "com.github.pureconfig"   %% "pureconfig"                   % "0.12.3",
   "uk.gov.hmrc"             %% "bootstrap-backend-play-26"    % "2.24.0",
-  "uk.gov.hmrc"             %% "simple-reactivemongo"         % "7.26.0-play-26"
+  "uk.gov.hmrc"             %% "simple-reactivemongo"         % "7.30.0-play-26",
+  "org.mongodb.scala"       %% "mongo-scala-driver"           % "2.9.0",
+  "org.mongodb"             %  "mongo-java-driver"            % "3.5.0"
```
Comment (MaciejRewera) [inline build.sbt:48]: Why are there both java and scala drivers?

File: precheck.sh [hunk 2]
```diff
@@ -0,0 +1,3 @@
+#!/usr/bin/env bash
+
+sbt clean coverage test it:test scalafmt::test test:scalafmt::test scalastyle coverageReport
```
Comment (MaciejRewera) [inline precheck.sh:3]: 👍

**Review by MaciejRewera — APPROVED** [review]

File: build.sbt (hunk 1 above)
Comment (glidester) [reply build.sbt:48]: The new migration code uses 'com.mongodb.client.MongoDatabase' from the java driver that I think mongock used to bring in as a dependency. I had a quick go at swapping to the scala driver in the code but due to slight differences in implementation of the drivers it was a non-trivial change so took more pragmatic approach. Maybe a tech debt job to migrate to scala driver in future :wink:

Comment (glidester) [reply build.sbt:48]: The new migration code uses 'com.mongodb.client.MongoDatabase' from the java driver that I think mongock used to bring in as a dependency. I had a quick go at swapping to the scala driver in the code but due to slight differences in implementation of the drivers it was a non-trivial change so took more pragmatic approach :)

**Pull request merged**